File: C++/level2/뉴스_클러스터링.cpp
#include <string>
#include <cctype>
#include <iostream>
#include <algorithm>

using namespace std;

void returnPices(vector<string> &pices, string str){
    for(int i=0; i<str.size()-1; i++){
        if(((str[i] >= 'a' && str[i] <= 'z') || (str[i] >= 'A' && str[i] <= 'Z')) &&
          ((str[i+1] >= 'a' && str[i+1] <= 'z') || (str[i+1] >= 'A' && str[i+1] <= 'Z'))){
            string temp = "";
            temp += toupper(str[i]);
            temp += toupper(str[i+1]);
            pices.push_back(temp);
        }
    }
}

int solution(string str1, string str2) {
    int answer = 0;
    vector<string> pices1;
    vector<string> pices2;

    returnPices(pices1, str1);
    returnPices(pices2, str2);

    sort(pices1.begin(), pices1.end());
    sort(pices2.begin(), pices2.end());

    int jVal = 0;
    for(int i=0; i<pices1.size(); i++){
        for(int j=jVal; j<pices2.size(); j++){
            if(!pices1[i].compare(pices2[j])){
                pices1[i] = "";
                pices2[j] = "";
                jVal = j+1;
                answer++;
                break;
            }
        }
    }

    if(pices1.size() == 0 && pices2.size() == 0)
        return 65536;

    return (double)answer / (double)(pices1.size() + pices2.size() - answer) * (double)65536;
}

File: JavaScript/level2/뉴스_클러스터링.js
function solution(str1, str2) {
    var answer = 0;
    let arr1 = [];
    let arr2 = [];
    let index1 = 0;
    let index2 = 0;
    
    str1 = str1.toUpperCase();
    str2 = str2.toUpperCase();

    for(let i=1; i<str1.length; i++)
        if(str1[i-1] >= "A" && str1[i-1] <= "Z" && str1[i] >= "A" && str1[i] <= "Z")
            arr1.push(str1.substring(i-1, i+1));
    
    for(let i=1; i<str2.length; i++)
        if(str2[i-1] >= "A" && str2[i-1] <= "Z" && str2[i] >= "A" && str2[i] <= "Z")
            arr2.push(str2.substring(i-1, i+1));
    
    arr1.sort();
    arr2.sort();

    while(index1 < arr1.length && index2 < arr2.length){
        if(arr1[index1] === arr2[index2]){
            index1++;
            index2++;
            answer++;
        }
        else if(arr1[index1] < arr2[index2])
            index1++;      
        else
            index2++;       
    }

    if(arr1.length === 0 && arr2.length === 0)
        return 65536;
    
    return Math.floor(answer/(arr1.length + arr2.length - answer)*65536);
}